sdk/storage/azure-storage-blob/tests/test_blob_client_async.py

```diff
@@ -548,4 +548,17 @@ def callback(response):
         custom_headers = {'User-Agent': 'customer_user_agent'}
         await service.get_service_properties(raw_response_hook=callback, headers=custom_headers)
 
+    @GlobalStorageAccountPreparer()
+    async def test_closing_pipeline_client(self, resource_group, location, storage_account, storage_account_key):
+        # Arrange
+
+        for client, url in SERVICES.items():
+            # Act
+            service = client(
+                self.account_url(storage_account.name, "blob"), credential=storage_account_key, container_name='foo', blob_name='bar')
+
+            # Assert
+            async with service:
+                assert hasattr(service, 'close')
+                service.close()
```

Comment:
Side note, the "with" call close too, so this test is technically more stresfull that it should be, it closes the service, and then exit the contextmanager and tries to re-close it.
It's nice to see that the second close doesn't break it, but it's not really the nominal test :).

For sake of clarity I would do a simple "create and and close" and keep your stress test "create, enter context manager, close, close context manager"